### PR TITLE
Add --prev-state-dir option

### PR DESCRIPTION
**Pull request by kkaarreell**
Introduces `--prev-state-dir` option so that user doesn't have to provide `--state-dir` manually and instead the most recent state-dir used within the same shell session is used.

### CODE REVIEW

File: newa/cli.py [hunk 1]
```diff
@@ -41,35 +41,55 @@
     )
 
 JIRA_NONE_ID = '_NO_ISSUE'
+STATEDIR_PARENT_DIR = '/var/tmp/newa'
```
Comment (happz) [inline newa/cli.py:44]: It should be set to a `Path` instance right here, to prevent any path-like string sneaking in.

File: newa/cli.py [hunk 2]
```diff
@@ -41,35 +41,55 @@
...
+STATEDIR_NAME_PATTERN = '^run-([0-9]+)$'
```
Comment (happz) [inline newa/cli.py:45]: I'd recommend making this a raw string. There are no regex features that would be affected, but it's a good habit to build.

File: newa/cli.py [hunk 3]
```diff
@@ -41,35 +41,55 @@
...
             c = int(r.group(1))
-            if c >= counter:
+            if use_ppid:
+                ppid_file = parent_dir / entry.name / ppid_filename
+                if ppid_file.exists() and c >= counter:
+                    counter = c
+            elif c >= counter:
                 counter = c + 1
-    return parent_dir / f'run-{counter}'
+    if counter > 0:
```
Comment (happz) [inline newa/cli.py:78]: `if counter:` is a more pythonic version, `0` is a false-ish value.

File: newa/cli.py (hunk 1 above)
Comment (kkaarreell) [reply newa/cli.py]: fixed

File: newa/cli.py (hunk 3 above)
Comment (kkaarreell) [reply newa/cli.py]: fixed

File: newa/cli.py (hunk 2 above)
Comment (kkaarreell) [reply newa/cli.py]: fixed

**Comment by kkaarreell** [comment]
I have realized the algorithm is not correct so I have modified it slightly. Also, I think it might be easier to understand now.

**Pull request merged**